Modules/Core/Common/include/itkSingleton.h

```diff
@@ -32,8 +32,7 @@
  * (i.e. itkNotUsed cannot be used) but is not always used.
  */
 template <typename T>
-[[deprecated("Preferably use the C++ `[[maybe_unused]]` attribute instead!")]] inline void
```

Comment:
This commit seems to be fighting against clang-format, see:
https://github.com/InsightSoftwareConsortium/ITK/actions/runs/9097194889/job/25004442774?pr=4665